chispa/number_helpers.py

```diff
@@ -8,5 +9,27 @@ def isnan(x):
         return False
 
 
-def nan_safe_equality(x, y) -> bool:
-    return (x == y) or (isnan(x) and isnan(y))
\ No newline at end of file
+def check_equal(
+    x, y,
+    precision: Optional[float] = None,
+    allow_nan_equality: bool = False,
+) -> bool:
+    """Return True if x and y are equal.
+
+    Parameters
+    ----------
+    precision : float, optional
+        Absolute tolerance when checking for equality.
+    allow_nan_equality: bool, defaults to False
+        When True, treats two NaN values as equal.
+
+    """
+    both_floats = (isinstance(x, float) & isinstance(y, float))
```

Comment:
Hi @MrPowers , can you have a look at [my draft pull request here](https://github.com/mitches-got-glitches/chispa/pull/3) to see if I'm going along the right lines with your 3rd objective?

To summarise, rather than the following line in `check_equal`:

```python
both_floats = (isinstance(x, float) & isinstance(y, float))
```

I'm doing this:

```python
is_float_type = (dtype_name in ['float', 'double', 'decimal'])
```

Where dtype gets passed into the function, and is created with the following line:

```python
dtypes = [field.dataType.typeName() for field in df1.schema]
```

Because we've already compared the schemas, we know that they're equal so we just use `df1.schema`.

Will using `isinstance` really have a noticeable effect on the speed of the tests? I feel like it might be the safer option.

Reply:
I just did some further testing too, and my current solution would not work for pyspark columns with `DecimalType`, as the python type is `decimal.Decimal` not `float`. I'd need to change that if the original implementation is selected.

Reply:
Any time to give me a steer this week @MrPowers ?